ci-bench/src/cachegrind.rs

```diff
@@ -291,3 +291,24 @@ pub fn diff(baseline: &Path, candidate: &Path, scenario: &str) -> anyhow::Result
 
     Ok(diff)
 }
+
+pub(crate) fn pause_instruction_count() -> PauseGuard {
+    PauseGuard::new()
+}
+
+pub(crate) struct PauseGuard;
```

Comment:
Maybe worth an internal rustdoc comment to warn this won't do the right thing (AFAICT) if you end up creating multiple `PauseGuard`'s and drop one but not the other. I think the first that gets dropped would toggle collection off and the next drop would turn it back on?

Probably not a concern for real usage :shrug: